src/region.js

```diff
@@ -204,12 +208,38 @@ Marionette.Region = Marionette.Object.extend({
     return Backbone.$(el, Marionette._getValue(this.options.parentEl, this));
   },
 
+  // Replace the region's DOM element with the view's DOM element.
+  replaceEl: function(view) {
+    // empty el so we don't save any non-destroyed views
+    this.$el.contents().detach();
+
+    var parent = this.el.parentNode;
+
+    parent.replaceChild(view.el, this.el);
+    this.replaced = true;
+  },
+
+  // Restore the region's element in the DOM.
+  restoreEl: function() {
+    var view = this.currentView;
+    var parent = view.el.parentNode;
+
+    parent.replaceChild(this.el, view.el);
+    this.replaced = false;
+  },
+
   // Override this method to change how the new view is
   // appended to the `$el` that the region is managing
-  attachHtml: function(view) {
-    this.$el.contents().detach();
+  attachHtml: function(view, shouldReplace) {
+    if (shouldReplace) {
+      // replace the region's node with the view's node
+      this.replaceEl(view);
+    } else {
+      // empty the node and append new view
+      this.$el.contents().detach();
```

Comment:
As above - detach is used in the current implementation of attachHtml I've just moved it to the else case here